Containerized Docker Application Lifecycle - 04.13 (#10617)
dotnet/docs: docs/standard/containerized-lifecycle-architecture/design-develop-containerized-apps/build-aspnet-core-applications-linux-containers-aks-kubernetes.md

description: Containerized Docker Application Lifecycle with Microsoft Platform and Tools
author: CESARDELATORRE
ms.author: wiwagn
ms.date: 02/15/2019
---
# Build ASP.NET Core 2.1 applications deployed as Linux containers into AKS/Kubernetes orchestrator

Azure Kubernetes Services (AKS) is Azure's managed Kubernetes orchestrations services that simplify container deployment and management.

AKS main features are, an Azure-hosted control plane, automated upgrades, self-healing, user configurable scaling, and a simpler user experience for both developers and cluster operators.

> In the following examples we'll explore the creation of an ASP.NET Core 2.1 application, using Visual Studio 2017 that runs on Linux and deploys to an AKS Cluster in Azure.

Author: This paragraph looked familiar to me! @mairaw and I had reviewed this article before and this change was reverting the previous review 😣

**Figure 4-37**. Selecting ASP.NET CORE 2.1 and Web API project type

If you have any previous version of .NET Core, you can download and install the 2.1 version from <https://www.microsoft.com/net/download/core#/sdk>

You can add Docker support when creating the project in the previous step, or later, so you can "Dockerize" your project any time. To add the Docker support after the project creation, right-click on the solution file and select **Add > Docker support** on the context menu.

![Context menu option to add Docker support to a existing project: Right click (on the project) > Add > Docker Support.](media/add-docker-support-to-project.png)

**Figure 4-38**. Adding Docker support to existing project

To complete adding Docker support, you have the choice of Windows or Linux, in this case select **Linux**, because AKS doesn’t support Windows Containers (as of late 2018).

![Option dialog to select Target OS for Dockerfile.](media/select-linux-docker-support.png)

**Figure 4-39**. Selecting Linux Containers.

With these simple steps, you will have your ASP.NET Core 2.1 application running on a Linux Container.

> As you can see the, integration between Visual Studio 2017 and Docker is totally oriented to the developer’s productivity.

Reviewer: ```suggestion
As you can see, the integration between Visual Studio 2017 and Docker is totally oriented to the developer’s productivity.
```

> Now you can run your application with the key **F5** or using the Play Button

Reviewer: ```suggestion
Now you can run your application with the **F5** key or by using the Play Button.
```

> After running the project, you can list the images using the `docker images` command, and you should see the `mssampleapplication` image created with the automatic deploy of our project with Visual Studio 2017.

Reviewer: ```suggestion
After running the project, you can list the images using the `docker images` command, and you should see the `mssampleapplication` image created with the automatic deployment of our project with Visual Studio 2017.
```

```console
# docker images
```

![Console output from the docker images command, shows a list with: Repository, Tag, Image Id, Created (date), and Size.](media/docker-images-command.png)

**Figure 4-40**. View of docker images

## Register the Solution in the Azure Container Registry

> Now, we have to upload the image to any Docker registry, like [Azure Container Registry (ACR)](https://azure.microsoft.com/services/container-registry/) or Docker Hub so the images can be deployed to the AKS cluster from that registry. In this case, we’re uploading the image to Azure Container Registry.

Reviewer: ```suggestion
Now, we have to upload the image to any Docker registry, like [Azure Container Registry (ACR)](https://azure.microsoft.com/services/container-registry/) or Docker Hub, so the images can be deployed to the AKS cluster from that registry. In this case, we’re uploading the image to Azure Container Registry.
```

### Create the image in Release mode

> We'll now create the image in **Release** Mode (ready for production) changing to Release as shown here and running the application as did before.

Reviewer: ```suggestion
We'll now create the image in **Release** mode (ready for production) by changing to **Release** as in Figure 4-41 and running the application as did we before.
```

![Toolbar option in VS to build in release mode.](media/select-release-mode.png)

**Figure 4-41**. Selecting Release Mode

> If you execute the `docker image` command, you'll see both images created one for `debug` and the other for `release` mode.

Reviewer: ```suggestion
If you execute the `docker image` command, you'll see both images created, one for `debug` and the other for `release` mode.
```

**Figure 4-43**. Get the name of the registry using PowerShell

In both cases, you'll obtain the name, in our example `mssampleacr.azurecr.io`

> Now you can Tag the image, taking the latest image (Release image), with the command:

Reviewer: ```suggestion
Now you can tag the image, taking the latest image (the Release image), with the command:
```

```console
docker tag mssampleaksapplication:latest mssampleacr.azurecr.io/mssampleaksapplication:v1
```

> After running the `docker tag` command, list the images with the `docker images` command, you should see the image with the new tag.

Reviewer: ```suggestion
After running the `docker tag` command, list the images with the `docker images` command, and you should see the image with the new tag.
```

**Figure 4-46**. View of nodes

> The next step is to deploy your container into your AKS Kubernetes cluster, for that you need a file (**.yml deploy file**) that, in this case, contains:

Reviewer: ```suggestion
The next step is to deploy your container into your AKS Kubernetes cluster. For that you need a file (**.yml deploy file**) that, in this case, contains:
```
